hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenCancelAirdropHandler.java

```diff
@@ -126,15 +134,127 @@ public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckExcept
         }
     }
 
+    @SuppressWarnings("java:S3864")
     @Override
     public void handle(@NonNull final HandleContext context) throws HandleException {
-        var tokensConfig = context.configuration().getConfigData(TokensConfig.class);
+        requireNonNull(context);
+        final var txn = context.body();
+        final var op = txn.tokenCancelAirdropOrThrow();
+        configValidation(context.configuration(), op);
+
+        final var accountStore = context.storeFactory().writableStore(WritableAccountStore.class);
+        final var airdropStore = context.storeFactory().writableStore(WritableAirdropStore.class);
+
+        final var pendingAirdropIds = op.pendingAirdrops();
+        final var payer = context.payer();
+        final var payerAccount = getIfUsable(payer, accountStore, context.expiryValidator(), INVALID_ACCOUNT_ID);
+        validateTrue(payerAccount.hasHeadPendingAirdropId(), INVALID_TRANSACTION_BODY);
```

Comment:
`SENDER_HAS_NO_AIRDROPS_TO_CANCEL` might be correct response code?

Reply:
Is this a new response code we need to add to the protos? Because I can't find it.

Reply:
yeah I mean that response would be appropriate. I can add it